packages/server/graphql/public/mutations/batchArchiveTasks.ts

```diff
@@ -18,22 +19,34 @@ const batchArchiveTasks: MutationResolvers['batchArchiveTasks'] = async (
 
   // VALIDATION
   const tasks = (await dataLoader.get('tasks').loadMany(taskIds)).filter(isValid)
-  const validTasks = tasks.filter(async ({createdBy, teamId}) => {
+  const validTasksByTeamId = {} as {[teamId: string]: Task[]}
+
+  for (const task of tasks) {
+    const {createdBy, teamId} = task
+    const tasks = validTasksByTeamId[teamId] || []
+
     if (createdBy === viewerId) {
       // if viewer is the task owner, they can archive
-      return true
+      tasks.push(task)
+    } else {
+      const teamMembers = await dataLoader.get('teamMembersByTeamId').load(teamId)
+      if (teamMembers.some(({userId}) => userId === viewerId)) {
```

Comment:
-1 this can be simplified to
```ts
if (viewer.tms.includes(teamId)) {
```
We can load the viewer in l21 in a `Promise.all`